go/vt/vttablet/tabletmanager/filesystemmanager.go

```diff
@@ -0,0 +1,119 @@
+package tabletmanager
+
+import (
+	"context"
+	"os"
+	"path"
+	"strconv"
+	"sync"
+	"time"
+
+	"vitess.io/vitess/go/vt/env"
+)
+
+type FileSystemManager interface {
+	// IsDiskStalled returns true if the disk is stalled or rejecting writes.
+	IsDiskStalled() bool
+}
+
+func newFileSystemManager(ctx context.Context) FileSystemManager {
+	if !enableStalledDiskCheck {
+		return newNoopFilesystemManager()
+	}
+
+	return newPollingFileSystemManager(ctx, attemptFileWrite, stalledDiskWriteInterval, stalledDiskWriteTimeout)
+}
+
+type writeFunction func() error
+
+func attemptFileWrite() error {
+	file, err := os.Create(path.Join(env.VtDataRoot(), ".stalled_disk_check"))
```

Comment:
Are there any cases where `VTDATAROOT` isn't the same disk as MySQL datadir? For our deploying it IS the same disk - our `datadir` is one level below `VTDATAROOT`. But theoretically `datadir` could be on it's own disk 🤔 

It might be more flexible to have a flag like `--stalled_disk_check_dir` / `stalled_disk_check_root` enable this feature vs a boolean "enable" flag